### ISSUE TITLE
[Benchmarks][Upstream PyTorch 2.5] `Triton` and `XeTLA` softmax performance degrades in comparison with  `torch 2.1` / `ipex 2.1` test proxies

### ISSUE THREAD

**Issue by ESI-SYD** [intel/intel-xpu-backend-for-triton#2106]
2. Ratio of `Triton`/ `XeTLA` keep same except for attention caused by `XeTLA` attention absolute number degraded
3. Both `Triton` and `XeTLA` softmax cases degraded, so `Triton`/ `XeTLA` not changed.

details: https://github.com/intel/intel-xpu-backend-for-triton/pull/1905#issuecomment-2320701513

**Comment by vlad-penkin** [comment]
@ESI-SYD what is the root cause for this issue? can you pin point it to a particular `torch` operation?

To proceed further with analysis / triaging please create a minimal reproducer for the Triton kernel path.

**Comment by ESI-SYD** [comment]
> @ESI-SYD what is the root cause for this issue? can you pin point it to a particular `torch` operation?
 
There are two main differences in benchmark time method change after applying the [Draft](https://github.com/intel/intel-xpu-backend-for-triton/pull/1905)

1. No sync submitting.  https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/python/triton/testing.py#L214

3. Use the time stamp between two barriers which is not accurate. Previous detailed [explanation](https://github.com/intel/intel-xpu-backend-for-triton/pull/1136/files#r1605626198) by chengjun.

**Comment by anmyachev** [comment]
https://github.com/intel/intel-xpu-backend-for-triton/pull/2149#issuecomment-2337632244

**Comment by anmyachev** [comment]
> 2. Ratio of Triton/ XeTLA keep same except for attention caused by XeTLA attention absolute number degraded

At the moment, the degradation of absolute numbers has been fixed. The geometric mean difference is ~2% (between [#1](https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10875041595/job/30173001940) and [#2](https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10881113472/job/30189366871)), which can be considered within the margin of error, I believe.
